### ISSUE TITLE
Assertions for numbers are type-specific, which can be too restrictive and slow development down

### ISSUE THREAD

**Issue by jamie-burns** [Bungalow64/DBConfirm#52]
Using v0.1.3, SQL Server and MSTest.

Say there is a stored procedure that returns a column that is set as smallint (e.g., Quantity in Northwind's dbo.CustOrderDetail).  I write a test that verifies that this number is 2.

If I write the assertion like this:

```csharp
result
  .AssertRowCount(1)
  .ValidateRow(0)
  .AssertValues(new DataSetRow
  {
      { "Quantity", 2 }
  });
```

Then the test fails, with this error:
> Assert.AreEqual failed. Expected:<2 (System.Int32)>. Actual:<2 (System.Int16)>. Column Quantity in row 0 has an unexpected value

I would have to cast the expected value to the correct type for the test to pass:

```csharp
result
  .AssertRowCount(1)
  .ValidateRow(0)
  .AssertValues(new DataSetRow
  {
      { "Quantity", (short)2 }
  });
```

This applies to all number types - for example decimals, which requires either casting or postfixing with M).

It would be helpful for some implicit casting, where possible.  If I write the number 2, then that should be able to be used in an assertion with any number type that can represent this number fully, without failing on a type check.  If a number is used that cannot be cast fully (e.g., a 1.5 used to assert a shortint column), then the original type needs to be used, since it wouldn't make sense to use either 1 or 2 for this assertion.

The only downside for relaxing the rules on type checking is that it's no longer possible (in 0.1.3) to write a test that locks down the type of data returned for a given column.  This is currently done by default when the value is asserted.  So it might be a good idea to add extra validation methods to explicitly check for the column type (e.g., `AssertColumnType(string expectedColumnName, Type expectedType)`), so that tests can still check this if required.

TL;DR;

1. When asserting numeric values, attempt to avoid unnecessary type errors (possibly by exact casting)
2. ~~Add extra validation methods to test for column types~~ (already possible)

**Comment by jamie-burns** [comment]
Note, it's already possible to do an explicit type check using an IComparison object, like this:

```csharp
result
    .ValidateRow(0)
    .AssertValue("Quantity", Comparisons.IsType(typeof(short)));
```

**Comment by jamie-burns** [comment]
As for comparing numbers, the documentation already shows that numeric assertions are type-specific (https://www.dbconfirm.com/writingtests, at the bottom - 'Asserting numbers').  It would be a breaking change to alter this behaviour by default, so an alternative solution would be to add a new IComparison object which loosely asserts numeric values.

**Comment by jamie-burns** [comment]
This has been merged into the develop branch, for v1.1.0. This issue will be updated once the changes have been deployed to NuGet.
